infrastructure/common-images/workspaces-base/backend/api/helpers/density_map/grid_creator.py

```diff
@@ -0,0 +1,145 @@
+
+    resolution_x = 1 / (bg_atlas.resolution[2] * UM_TO_MM)
+    resolution_y = 1 / (bg_atlas.resolution[1] * UM_TO_MM)
+    resolution_z = 1 / (bg_atlas.resolution[0] * UM_TO_MM)
+
+    # calculates ratio from image size to atlas size
+    ratio_x = w / x_shape
+    ratio_y = h / y_shape
+
+    canal_offset_x *= resolution_x
+    canal_offset_y *= resolution_y * -1
+
+    x_ticks, x_ticks_labels = _get_ticks(canal_offset_x, ratio_x, x_shape)
+    y_ticks, y_ticks_labels = _get_ticks(canal_offset_y, ratio_y, y_shape)
+
+    ax.set_xticks(x_ticks)
+    ax.set_yticks(y_ticks)
+    ax.set_xticklabels(x_ticks_labels)
+    ax.set_yticklabels(y_ticks_labels)
+
+    ax.set_title(f'{_get_subdivision_depth(bg_atlas, subdivision, resolution_z)} mm',
+                 fontdict={'fontsize': 8,
+                           'fontweight': rcParams['axes.titleweight'],
+                           'color': GRID_COLOR,
+                           'verticalalignment': 'baseline',
+                           'horizontalalignment': 'right'},
+                 loc='right')
+
+    # Sets axis color
+    ax.spines['bottom'].set_color(GRID_COLOR)
+    ax.spines['top'].set_color(GRID_COLOR)
+    ax.spines['left'].set_color(GRID_COLOR)
+    ax.spines['right'].set_color(GRID_COLOR)
+    ax.xaxis.label.set_color(GRID_COLOR)
+    ax.yaxis.label.set_color(GRID_COLOR)
+    ax.tick_params(axis='x', colors=GRID_COLOR)
+    ax.tick_params(axis='y', colors=GRID_COLOR)
+
+    for item in ([ax.title, ax.xaxis.label, ax.yaxis.label] +
+                 ax.get_xticklabels() + ax.get_yticklabels()):
+        item.set_fontsize(8)
+
+    plt.grid(False)
+    img = fig_to_img(fig)
+
+    # Corrects image alignment by adding a constant amount of padding to the image
+    shifted_image = _add_margin(img, 0, CONSTANT_RIGHT_OFFSET, CONSTANT_BOTTOM_OFFSET, 0)
+    return shifted_image
+
+
+def _get_ticks(canal_offset, img_to_res_ratio, shape, step=0.1):
+    # ticks are on image scale
+    ticks = np.round(
+        np.arange(
+            _get_tick_rounded(canal_offset * -1 - shape / 2, img_to_res_ratio),
+            _get_tick_rounded(canal_offset * -1 + shape / 2 + step, img_to_res_ratio), step),
+        2)
+    # tick labels apply the canal center offset
+    ticks_labels = []
+    for idx, tick in enumerate(ticks):
+        if idx % 5 == 0:
+            label = round((tick - canal_offset * -1) * img_to_res_ratio, 2)
+            if round(label, 1) * 10 == 0:
+                label = 0
+            ticks_labels.append(label)
+        else:
+            ticks_labels.append('')
+    return ticks, ticks_labels
+
+
+def _get_tick_rounded(original, ratio):
+    # gets tick value in the image scale so that it starts with 0 on the 2 decimal place
+    scaled = original * ratio
+    diff = round(round(scaled, 2) - round(scaled, 1), 2)
+    return original + diff / ratio
+
+
+def _set_size(w, h, ax=None) -> (float, float):
+    """ w, h: width, height in inches """
+    if not ax: ax = plt.gca()
```

Comment:
did you run the black python linter?

Reply:
No, I will. Maybe we should have a pre-commit hook to force me lint?